docs/builtin-objects.md

```diff
@@ -0,0 +1,52 @@
+# environment template built-in objects
+
+- `Environment`: The information about the environment. This is set by the
+  `--environment` flag. It has several objects inside of it:
+  - `Environment.Name`: The name of the environment
+- `StateFile`: The information about the state file. It has several objects
+  inside of it:
+  - `StateFile.Name`: The name of the current state file
+  - `StateFile.BasePath`: The base path of the current state file
+  - `StateFile.RootPath`: The path of the root state file
```

Comment:
I think we'd better name it:

```suggestion
  - `RootStateFile.Path`: The path of the root state file
```

along with RootStateFile.Name and RootStateFile.BasePath if we are to expose those fields, for consistency.

Reply:
What is the expected use-case of the root path?